src/main/java/com/redhat/devtools/intellij/telemetry/core/service/UserId.java

```diff
@@ -60,12 +70,26 @@ private String load(Path uuidFile) {
         return uuid;
     }
 
-    private void write(String uuid, Path uuidFile) {
+    private boolean isValid(String uuid) {
+        if (uuid == null) {
+            return false;
+        }
+        return UUID_REGEX.matcher(uuid).matches();
+    }
+
+    private String create() {
+        return UUID.randomUUID().toString();
+    }
+
+    /** for testing purposes */
+    protected String write(String uuid, Path uuidFile) {
         try {
             FileUtils.createFileAndParent(uuidFile);
             FileUtils.write(uuid, uuidFile);
+            return uuid;
```

Comment:
why return the passed uuid here ? 
doesn't make difference to keep same signature as before.